File: subarray_palindrome.py
def get_subarrays(array):
    """Find all subarrays"""
    subarrays = []
    n = len(array)
    for i in range(n):
        for j in range(i+1, n+1):
            subarrays.append(array[i:j])
    return subarrays

def check_palindrome(array):
    """Checks if array is palindrome"""
    if not len(array)%2==0:
        del array[len(array)//2]
    for i in range(len(array)//2):
        if not array[i]==array[-i-1]:
            return False
    return True

def main():
    n = int(input()) # length of the array
    num_str = input()
    A = [] # The input array
    for n in num_str.split()[:n]:
        A.append(int(n))
    palindromes = 0
    subarrays = get_subarrays(A)
    for subarray in subarrays:
        palindromes += 1 if check_palindrome(subarray) else 0
    print(palindromes)
        

if __name__ == '__main__':
    try:
        main()
    except Exception as e: 
        print('error: ', e)

File: subarray_palindrome.c
#include<stdio.h>
#include<stdlib.h>

void main(){
    // get array size
    short N;
    int palindromes = 0;
    scanf("%hi", &N);

    // input array elements
    unsigned int *A = malloc(N * sizeof(unsigned int));
    for(short i = 0; i<N; i++){
        scanf("%u", &A[i]);
    }

    // find subarrays
    int x;
    for(short i = 0; i<N; i++){
        for(short j=i+1; j<=N; j++){
            short is_palindrome=1;

            //start of subarray is i and end is j-1
            x = 0;
            for(int k=i; k<=(j-1 + i)/2 ; k++){
                if(A[k]!=A[j-1-x]){
                    is_palindrome=0;
                    break;
                }
                x++;
            }
            if(is_palindrome==1) palindromes+=1;
        }
    }

    printf("%d\n", palindromes);
}